src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,172 @@
+    void setUp() {
+        user = new User();
+        user.setAge(18);
+        user.setLogin("testLogin");
+        user.setPassword("testPassword");
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user.setPassword(null);
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ZeroCharactersLogin_notOk() {
+        user.setLogin("");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ThreeCharactersLogin_notOk() {
+        user.setLogin("abc");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_FiveCharactersLogin_notOk() {
+        user.setLogin("abcde");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_SixCharactersLogin_Ok() {
+        user.setLogin("abcdef");
+        registrationService.register(user);
+        Assertions.assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_EightCharactersLogin_Ok() {
+        user.setLogin("abcdefgh");
+        registrationService.register(user);
+        Assertions.assertTrue(Storage.people.contains(user));
+    }
+
+    @Test
+    void register_ZeroCharactersPassword_notOk() {
+        user.setPassword("");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_ThreeCharactersPassword_notOk() {
+        user.setPassword("abc");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_FiveCharactersPassword_notOk() {
+        user.setPassword("abcde");
+        Assertions.assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_SixCharactersPassword_Ok() {
```

Comment:
the same: maybe you should replace word "six" into "minLength" or something like this. But it's up to you